core/vm/gas_table.go

```diff
@@ -439,20 +441,25 @@ func gasStaticCall(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memo
 
 func gasSelfdestruct(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
 	var gas uint64
+	contractAddr, err := contract.Address().InternalAddress()
+	if err != nil {
+		return 0, err
+	}
 	// EIP150 homestead gas reprice fork:
 	if evm.chainRules.IsEIP150 {
 		gas = params.SelfdestructGasEIP150
-		var address = common.Address(stack.Back(0).Bytes20())
-		exist, err := evm.StateDB.Exist(address)
+
+		address, err := common.Bytes20ToAddress(stack.Back(0).Bytes20()).InternalAddress()
+		exist, _ := evm.StateDB.Exist(*address)
```

Comment:
no need to catch error?

Reply:
https://github.com/dominant-strategies/go-quai/commit/cd76687c6060f9f9965e69237885eb6221754558 reverts this so no error is returned